From 81cb0feb1fca83c48e3ace4264abdfda18b7f3ba Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Sun, 15 Feb 2015 16:59:24 +0800
Subject: [PATCH 273/273] atomisp2: W/A: isp pad_num to 12x12 for ov2722 &
 1080p

ov2722 output resolution is 1932x1092, so need change
isp crop padding to 12x12

Change-Id: Id01ef20b8b7ff8d7dcb5cbf710d329d6df9495d1
Tracked-On: IMINAN-11434
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/41707
Reviewed-by: Jing Tao <jing.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c  | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
index 0f74b9e..ccefdca 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_subdev.c
@@ -399,7 +399,8 @@ int atomisp_subdev_set_selection(struct v4l2_subdev *sd,
 
 		/* Workaround for BYT 1080p perfectshot since the maxinum resolution of
 		 * front camera ov2722 is 1932x1092 and cannot use pad_w > 12*/
-		if (IS_BYT && crop[pad]->height == 1092) {
+		if (!strncmp(isp->inputs[isp_sd->input_curr].camera->name,
+				"ov2722", 6) && crop[pad]->height == 1092) {
 			padding_w = 12;
 			padding_h = 12;
 		}
-- 
1.9.1

